packages/twenty-front/src/modules/object-record/record-group/hooks/useRecordGroupSelector.ts

```diff
@@ -0,0 +1,110 @@
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import { useRecoilComponentCallbackStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackStateV2';
+import { usePersistViewGroupRecords } from '@/views/hooks/internal/usePersistViewGroupRecords';
+import { useGetViewFromCache } from '@/views/hooks/useGetViewFromCache';
+import { currentViewIdComponentState } from '@/views/states/currentViewIdComponentState';
+import { ViewGroup } from '@/views/types/ViewGroup';
+import { useRecoilCallback } from 'recoil';
+import { v4 } from 'uuid';
+import { isUndefinedOrNull } from '~/utils/isUndefinedOrNull';
+
+type UseRecordGroupSelectorParams = {
+  viewBarComponentId: string;
+};
+
+export const useRecordGroupSelector = ({
```

Comment:
Selector does not feel to be the right naming, also here we are reading from view which seems to be what record-index should do.
record-group should not take dependency on view to me. Record-index is the "glue" between view and record-groups